### PR TITLE
test_optim: use individual optimizer's defaults

### PR DESCRIPTION
**Pull request by skeydan**
Hi,

I think that as a user, if I don't  pass a learning rate via `opt_hparams`, I expect `test_optim()` to instantiate the optimizer with its default learning rate. 
With an optimizer like `optim_adam(),` I probably wouldn't have noticed, but for `optim_adahessian()`, the difference between it's, and the function's, default is enormous.

My code presupposes that every `optim_x()` will have a default learning rate. What do you think @dfalbel, is this reasonable, and if not, do you have a suggestion?

### CODE REVIEW

File: R/utils-testopt.R [hunk 1]
```diff
@@ -210,7 +210,10 @@ test_optim <- function(optim, ...,
     y <- torch::torch_tensor(y0, requires_grad = TRUE)
 
     # instantiate optimizer
-    optim <- do.call(optim, c(list(params = list(x, y)), opt_hparams))
+    if(!missing(opt_hparams))
```
Comment (dfalbel) [inline R/utils-testopt.R]:
```suggestion
    if(!is.null(opt_hparams))
```

`missing` checks if the argument has been specified or not. Since there's a default of `NULL`, I think it will always say `FALSE`.

File: R/utils-testopt.R [hunk 2]
```diff
@@ -155,7 +155,7 @@ domain_sphere <- function(){
 #'
 #' @export
 test_optim <- function(optim, ...,
-                       opt_hparams = list(lr = 0.01),
+                       opt_hparams = NULL,
```
Comment (dfalbel) [inline R/utils-testopt.R]:
```suggestion
                       opt_hparams = list(),
```

Then you wouldn't need to check if it's `NULL` or not below.

**Review by dfalbel — COMMENTED** [review]
Just made a few comments. I'm not sure all optimizers have a default learning rate, but I guess it's fine that in this case the user is required to specify it.

File: R/utils-testopt.R (hunk 1 above)
Comment (skeydan) [reply R/utils-testopt.R]: Well, i tested for that:

```
f <- function(x = NULL) {      
  return(missing(x))
}

f()

1] TRUE

```
